### PR TITLE
Remove all dead assignment detected by clang static analysis

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
Remove all dead assignment detect by clang static analysis.

**Test plan**
Clang ci is broken so...

```
scan-build "meson build && meson compile -C build"
scan-view <report>
```

**Closing issues**
...

### CODE REVIEW

File: binrz/rizin/rizin.c [hunk 1]
```diff
@@ -12,15 +12,12 @@ static void rz_cmd(int in, int out, const char *cmd) {
                 return;
         }
         while (1) {
-                int n = read (in, buf, bufsz);
```
Comment (ret2libc) [inline binrz/rizin/rizin.c:15]: Why did you just remove this read?

Comment (ghost) [reply binrz/rizin/rizin.c:15]: good question

**Comment by ghost** [comment]
binrz/rizin/rizin.c is ugly because the indentation was invalid (using space and no tab)

File: binrz/rizin/rizin.c [hunk 2]
```diff
@@ -4,53 +4,51 @@
...
-        write (out, cmd, strlen (cmd) + 1);
-        write (out, "\n", 1);
-        int bufsz = (1024 * 64);
-        unsigned char *buf = malloc (bufsz);
-        if (!buf) {
-                return;
-        }
-        while (1) {
-                int n = read (in, buf, bufsz);
-				buf[bufsz - 1] = '\0';
```
Comment (ret2libc) [inline binrz/rizin/rizin.c:16]: why was this removed? I think it is important to put a 0 at  the end of the buffer, no?

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]
